rhombus-lib/rhombus/private/amalgam/rx_object.rhm

```diff
@@ -0,0 +1,189 @@
+#lang rhombus/private/amalgam/core
+import:
+  lib("racket/base.rkt")
+  "core-meta.rkt" open
+  "maybe.rhm" open
+  "recur.rhm" open
+  "rx_match.rhm" open
+
+export:
+  RX
+  _RX
+  splice_regexp
+  freeze
+
+expr.macro 'MATCH $body ...':
+  'match $body ...'
+
+class RX(private _rkt_partial_rx,
+         private _rkt_rx,
+         private _num_captures :~ Int,
+         private _vars :~ Map,
+         private _has_backref,
+         private _str):
+  internal _RX
+  constructor (): error("don't instantiate")
+
+  private implements Printable
+  private override describe(mode, recur) :: String:
+    _str
+
+  property handle: _rkt_rx
+  property in_handle: _rkt_partial_rx
```

Comment:
Are these properties meant to be public?  If so, they should be documented.

Reply:
Yes, I forgot to document these. I also meant to add `RX.from_handles`.